src/Tenancy/Identification/TenantResolver.php

```diff
@@ -153,7 +154,13 @@ protected function resolveFromDrivers(TenantModelCollection $models): ?Tenant
                 foreach ($drivers as $driver) {
                     /** @var ReflectionMethod $method */
                     foreach ($this->retrieveDriverMethods($driver) as $method) {
-                        if ($tenant = app()->call("$item@{$method->getName()}")) {
+                        try {
+                            $tenant = app()->call("$item@{$method->getName()}");
+                        } catch (BindingResolutionException $e) {
+                            // Prevent trying to find a tenant when bindings aren't working for them.
+                        }
+
+                        if ($tenant) {
```

Comment:
Uhmmm this might actually be something that we want to add no matter what, if this PR is not going through, please make this a separate PR.

Reply:
This code is very specific to this functionality, will separate it if this PR isn't merged.